libshaderc_util/src/compiler.cc

```diff
@@ -209,15 +220,13 @@ bool Compiler::Compile(
       *error_stream << "shaderc: internal error: compilation succeeded but "
                        "failed to disassemble: "
                     << text_or_error << "\n";
-      return false;
+      return std::make_tuple(false, std::vector<uint32_t>(), 0u);
     }
-    return shaderc_util::WriteFile(output_stream, text_or_error);
+    return std::make_tuple(true, ConvertStringToVector(text_or_error),
+                           text_or_error.size());
   } else {
-    return shaderc_util::WriteFile(
-        output_stream,
-        string_piece(
-            reinterpret_cast<const char*>(spirv.data()),
-            reinterpret_cast<const char*>(&spirv.back()) + sizeof(uint32_t)));
+    return std::make_tuple(true, std::move(spirv),
```

Comment:
Sorry to raise this now.  But does this really save the copy for the case of a binary SPIR-V module?  I'm concerned because you've populated the spirv vector, but the only copy-elision that is done is on the tuple, not the spirv vector.
It seems that to get copy elision you'd have to have made a std::tuple variable in the first place where you currently have the spirv variable definition.